spras/analysis/summary.py

```diff
@@ -44,28 +47,43 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame) ->
         number_nodes = nw.number_of_nodes()
         number_edges = nw.number_of_edges()
         ncc = nx.number_connected_components(nw)
+
         # Initialize list to store current network information
         cur_nw_info = [nw_name, number_nodes, number_edges, ncc]
+
         # Iterate through each node property and save the intersection with the current network
         for node_list in nodes_by_col:
             num_nodes = len(set(nw).intersection(node_list))
             cur_nw_info.append(num_nodes)
+
+        # String split name to access algorithm and hashcode from filepath
+        # Name of filepath follows format "output/.../data#-algo-params-hashcode/pathway.txt"
+        # algorithm parameters have format { algo : { hashcode : { parameter combos } } }
+        filename = sorted(algo_with_params)[index].split("-")
+        algo = filename[0]
+        hashcode = filename[2]
+        index = index + 1
+
+        param_combo = algo_params[algo][hashcode]
+        params = json.dumps(param_combo)
+        params = params.replace("\"", "") #removes extra double quotes from string
+        cur_nw_info.append(params)
+
+        # Prepare column names
+        col_names = ["Name", "Number of nodes", "Number of undirected edges", "Number of connected components"]
```

Comment:
Should the header be updated to “Number of edges" instead of "Number of undirected edges"?

Even though we’re treating all graphs as undirected (on line 43), directionality doesn’t seem to be considered in any of the statistics being calculated in summary.py at the moment.

Reply:
I believe the original intention of this header was to convey that even a directed or mixed graph would be parsed and summarized as an undirected graph here. I can see how that could instead be interpreted as only counting the undirected edges in a mixed graph though. Which do you think is more precise?

Reply:
I was looking at the summary file and got confused on why were only counting the undirected edges. So I think renaming it to "Number of edges" is more intuitive and putting a comment about line 43 to say that "directed or mixed graph are parsed and summarized as an undirected graph"